corehq/apps/ota/views.py

```diff
@@ -310,19 +310,20 @@ def _get_version_info(app_config):
             "latest_ccz_version": config.get_latest_app_version(build_profile_id),
         }
 
+    app = get_app_cached(domain, app_build_id)
     try:
         version_info = _get_version_info(config)
         # mobile will send master app_id
     except (Http404, AssertionError):
         # If it's not a valid master app id, find it by talking to couch
         notify_exception(request, 'Received an invalid heartbeat request')
-        app = get_app_cached(domain, app_build_id)
         config = GlobalAppConfig.by_app_id(domain, app.master_id)
         version_info = _get_version_info(config)
 
     info.update(version_info)
 
-    if not toggles.SKIP_UPDATING_USER_REPORTING_METADATA.enabled(domain):
+    error_in_access = check_app_access(domain, request.couch_user, app)
```

Comment:
nit: error_msg, _ = check_app_access(domain, request.couch_user, app)

that way you can avoid the `[0]` later

Reply:
We can confirm with mobile team, I think we should give back the same response as we did for restore.
If someone does not have access to the app, we should not be doing anything at all and return back the appropritate response.

Reply:
Be careful using `_` as a throwaway variable name; this file uses it as an alias for `ugettext`.

Reply:
I have pinged on the ticket based on the response from @shubham1g5 I will make the required changes.